tlatools/org.lamport.tlatools/src/tla2sany/modanalyzer/SpecObj.java

```diff
@@ -288,19 +287,11 @@ private final ParseUnit findOrCreateParsedUnit(String name, Errors errors, boole
             // NamedInputStream for it if possible
             // SZ 23.02.2009: split the name resolution from the stream retrieval
             // NamedInputStream nis = this.ntfis.toNIStream(name);
-            NamedInputStream nis = FileUtil.createNamedInputStream(name, this.resolver);
-            
-			if (nis == null && rootParseUnit != null && rootParseUnit.getNis() != null
-					&& rootParseUnit.getNis().sourceFile() != null) {
-				// Fall back and try loading the module from a monolithic spec (one big .tla
-				// file consisting of multiple TLA+ modules and TLC configs) that is what is the
-				// rootModule here (compare tlc2.tool.impl.ModelConfig.parse()).
-            	try {
-					final File monolithSpec = rootParseUnit.getNis().sourceFile();
-					nis = MonolithSpecExtractor.module(monolithSpec, name);
-            	} catch (IOException e) {
-            		nis = null;
-            	}
+            NamedInputStream nis;
+            if (rootParseUnit != null) {
+                nis = FileUtil.createNamedInputStream(name, this.resolver, rootParseUnit.getNis());
+            } else {
+                nis = FileUtil.createNamedInputStream(name, this.resolver);
```

Comment:
I'am maintaining `rootParseUnit` handling here because `FileUtil` should know nothing about parsing specific stuff.